alphabet/alphabet_contract.go

```diff
@@ -33,19 +33,21 @@ func OnNEP17Payment(from interop.Hash160, amount int, data interface{}) {
 }
 
 func _deploy(data interface{}, isUpdate bool) {
-	if isUpdate {
-		return
-	}
-
 	args := data.(struct {
 		notaryDisabled bool
 		addrNetmap     interop.Hash160
 		addrProxy      interop.Hash160
 		name           string
 		index          int
 		total          int
+		version        int
```

Comment:
In initial deploy, `args` most likely won't have version field. Is it safe for VM to cast 6-element array into 7-element struct?

Reply:
Also I can imagine the case when `data interface{}` during contract update would provide additional information which is not related to contract deploy arguments. 

I thought that the robust way to fetch version is to manually fetch the last element of `[]interface` in `isUpdate` branch. That is the only reliable property of `data interface{}` during contract update.

```go
if isUpdate {
  arr := args.([]interface{})
  ln := len(arr)
  var version int = arr[ln-1]
  common.CheckVersion(version)
}

Reply:
>Is it safe for VM to cast 6-element array into 7-element struct?

Yes, both array and struct types are unsized.